src/main/java/net/wasdev/wlp/common/plugins/util/DevUtil.java

```diff
@@ -0,0 +1,956 @@
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.StandardCopyOption;
+import java.nio.file.StandardWatchEventKinds;
+import java.nio.file.WatchEvent;
+import java.nio.file.WatchKey;
+import java.nio.file.WatchService;
+import java.nio.file.Watchable;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Deque;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Scanner;
+import java.util.Set;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.jar.Attributes;
+import java.util.jar.JarFile;
+import java.util.jar.Manifest;
+
+import javax.tools.JavaCompiler;
+import javax.tools.JavaFileObject;
+import javax.tools.StandardJavaFileManager;
+import javax.tools.StandardLocation;
+import javax.tools.ToolProvider;
+
+import org.apache.commons.io.FileUtils;
+
+import com.sun.nio.file.SensitivityWatchEventModifier;
+
+/**
+ * Utility class for dev mode.
+ */
+public abstract class DevUtil {
+
+    /**
+     * Log debug
+     * 
+     * @param msg
+     */
+    public abstract void debug(String msg);
+
+    /**
+     * Log debug
+     * 
+     * @param msg
+     * @param e
+     */
+    public abstract void debug(String msg, Throwable e);
+
+    /**
+     * Log debug
+     * 
+     * @param e
+     */
+    public abstract void debug(Throwable e);
+
+    /**
+     * Log warning
+     * 
+     * @param msg
+     */
+    public abstract void warn(String msg);
+
+    /**
+     * Log info
+     * 
+     * @param msg
+     */
+    public abstract void info(String msg);
+
+    /**
+     * Log error
+     * 
+     * @param msg
+     */
+    public abstract void error(String msg);
+
+    /**
+     * Returns whether debug is enabled by the current logger
+     * 
+     * @return whether debug is enabled
+     */
+    public abstract boolean isDebugEnabled();
+
+    /**
+     * Stop the server
+     */
+    public abstract void stopServer();
+
+    /**
+     * Starts the default server
+     */
+    public abstract void startServer();
+
+    /**
+     * Updates artifacts of current project
+     */
+    public abstract void getArtifacts(List<String> artifactPaths);
+
+    /**
+     * Recompile the build file
+     * 
+     * @param buildFile
+     * @param artifactPaths
+     * @param executor The thread pool executor
+     * @return true if the build file was recompiled with changes
+     */
+    public abstract boolean recompileBuildFile(File buildFile, List<String> artifactPaths, ThreadPoolExecutor executor);
+
+    /**
+     * Get the number of times the application updated message has appeared in the application log
+     * 
+     * @return 
+     */
+    public abstract int countApplicationUpdatedMessages();
```

Comment:
If moving start and stop to common, I would also expect this method to be implemented in common as well, with the necessary parameters such as the artifact ID and log file passed in.

Reply:
Addressed above, we can open an issue to track using Ant directly at the common level.  Until that issue is resolved I think it is best to leave this as an abstract method for now.

Reply:
Moved to common with the following commits:
https://github.com/WASdev/ci.common/pull/48/commits/23d18416f984b2112956ab39f66982ec1925e752
https://github.com/WASdev/ci.maven/pull/476/commits/fcc8ea545cc8a419be9877be989f653a84725799